### PR TITLE
Improved repository searching

### LINKED ISSUE #5: Create a script to collect repositories provided a given language
One feature I would really like to see is the ability to follow repos provided a given language. For example, the following script would follow as many repositories as possible that use the Java language:

```
python3 follow_repos_by_language.py Java
```

Lingering questions:
- Because this can result in at least a thousand repos to be followed (I think that's what Github API returns to us, will need to double check), do we need to worry about throttling or forbidden responses after a certain number of requests? I don't have much experience running these scripts for LOTS of repositories so I don't have that answer.
- Is this a worthwhile issue to resolve?

### PR DESCRIPTION
**Pull request by mrthankyou**
## What Does This PR Do

This PR introduces two new scripts. These two scripts search and follow repositories based on conditions provided by the user:

1. *follow_repos_by_search_term.py*

    This script takes a search term and a programming language as input. It then finds repositories that contain the search term and the programming language used. When we find a repository, we then follow it on LGTM.com.

2. *follow_top_repos_by_star_count.py*

    This script takes a programming language as input. It then finds repositories that contain more than 500 stars and uses the programming language provided in the input.

## Issue Resolved
Resolves #5.

## Anything Else We Should Know

- These scripts all contain a simple throttle on HTTP requests. This throttle feature is used to ensure that users do not send too many requests at once.
- ~~I've tested both scripts on my local machine. Both scripts ran successfully.~~ I've made small adjustments to the code since my last test. As soon as I get a chance I'll re-run the scripts. From a quick look, I don't see any errors.
- I've structured the code so that once a Github repo is found, we immediately add it follow it on LGTM. I think this is a better approach than first collecting all Github repos and then following them on lgtm.com because if an internet connection breaks midway through the script running, we may never have actually followed any repos on LGTM. All that work would have been lost. Here at least we have some results than no results at all.
- I've introduced a new technique in searching for repos in these scripts. By default, Github will only return the first 1,000 results in any search query. To handle this limitation, we limit searches to specific date-times for when the repo was first created. This allows us to retrieve almost all repositories that match the query.*

#### Footnotes

\* This is easier to explain with the following example. Let's say we want to find repositories for a given programming language that have more than 500 stars. Normally, Github only retrieves the first 1,000 results. We can break this query up into multiple queries like so:

1. Find repositories with more than 500 stars for a given programming language where the repository was created between 2008 - 2009
2. Find repositories with more than 500 stars for a given programming language where the repository was created between 2009 - 2010
3. Find repositories with more than 500 stars for a given programming language where the repository was created between 2010 - 2011
4. Etc.

Even though each query is limited to the first 1,000 results, by breaking up the queries we are much more likely to get all results instead of the first 1,000 repositories created from 2008 till now.

### CODE REVIEW

File: follow_repos_by_search_term.py
Comment (JLLeitschuh) [inline follow_repos_by_search_term.py]: Can we extract this to a common script?

**Review by JLLeitschuh — CHANGES_REQUESTED** [review]
LGTM! Just one suggested change

File: follow_repos_by_search_term.py
Comment (mrthankyou) [reply follow_repos_by_search_term.py]: Done! I also extracted the date logic into a separate script and created a `utils` folder to store utility-related scripts.

**Comment by mrthankyou** [comment]
@JLLeitschuh 

Ready for re-review. Manually tested each script and it's running as expected!

**Comment by JLLeitschuh** [comment]
Can you fix the flagged unused imports by the LGTM bot, once resolved, I'm happy to merge this!

**Comment by JLLeitschuh** [comment]
TYVM!

**Pull request merged**
